io/include/detray/io/backend/detail/grid_reader.hpp

```diff
@@ -60,15 +60,25 @@ class grid_reader {
             &det_builder,
         const detector_grids_payload<content_t, grid_id_t> &grids_data) {
 
+        DETRAY_DEBUG("Generic grid reader from_payload: content_t="
+                     << DETRAY_TYPENAME(content_t)
+                     << ", grid_id_t=" << DETRAY_TYPENAME(grid_id_t));
+
         // Convert the grids volume by volume
+        DETRAY_DEBUG("Converting grids for " << grids_data.grids.size()
+                                             << " volumes");
         for (const auto &[_, grid_data_coll] : grids_data.grids) {
             for (const auto &[i, grid_data] :
                  detray::views::enumerate(grid_data_coll)) {
 
+                DETRAY_DEBUG("- volume #" << i);
```

Comment:
I think `i` just enumerates the json payloads
```suggestion
                DETRAY_DEBUG("- reading grid #" << i << " in volume #" << grid_data.owner_link.link);
```